snapshot/db.go

```diff
@@ -0,0 +1,45 @@
+package snapshot
+
+// ID identifies a value in DB
+type ID string
+
+// DB is the Scoot Database. It holds Values identified by an ID.
+// A Value may be a Snapshot or a Revision.
+type DB interface {
+	// Ingest
+
+	// IngestDir ingests a directory directly.
+	// The created value is a Snapshot.
+	IngestDir(dir string) (ID, error)
+
+	// IngestGitCommit ingests the commit identified by commitish from ingestRepo
+	// commitish may be any string that identifies a commit
+	// The created value is a Revision.
+	IngestGitCommit(ingestRepo *repo.Repository, commitish string) (SnapID, error)
+
+	// Operations
+
+	// SnapshotForRevision unwraps a Revision and returns a Snapshot.
+	// Errors if id does not identify a Revision.
+	SnapshotForRevision(id ID) (ID, error)
+
+	// Distribute
+
+	// Upload makes sure the value id is uploaded, returning an ID that can be used
```

Comment:
It persists them to a global central store; the details are hidden behind the interface.

I think for V1, we can have two places:
*) local
*) global

This puts a high bar on the service (we need global consistency), but allows us to work on your macbook while offline. WDYT?